### PR TITLE
Document file destination of build scripts.

### LINKED ISSUE #2754: docs should mention output file for build scripts
I was completely baffled by the lack of any output of the following build script:

```rust
fn main() {
    println!("Downloading and building CLDR data...");
    Command::new("python").args(&["scripts/download_import_cldr.py"])
                          .output()
                          .unwrap_or_else(|e| {
                              panic!("failed to execute process: {}", e)
                          });
    println!("Done!");
}
```

Cargo documentation for build scripts says nothing about output except that prefixing any output with "cargo:" lets you perform pass back metadata to Cargo.

I asked in IRC and was told the output of my script was written into `target/debug/build/<pkg>/output`.

A short blurb in the docs would be sufficient to mitigate confusion.

### PR DESCRIPTION
**Pull request by andrehjr**
Close #2754

### CODE REVIEW

File: src/doc/build-script.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ the source directory of the build script’s package.
 ## Outputs of the Build Script
 
 All the lines printed to stdout by a build script that start with `cargo:`
-are interpreted by Cargo and must be of the form `key=value`.
+are interpreted by Cargo and must be of the form `key=value`. Output is written to `target/debug/build/<pkg>/output`.
```
Comment (alexcrichton) [inline src/doc/build-script.md]: Could this elaborate a bit perhaps along the lines of:

> All the lines printed to stdout by a build script are written to a file like `target/debug/build/<pkg>/output` (the precise location may depend on your configuration), and any line that starts with `cargo:` is interpreted directly by Cargo. This line must be of the form `cargo:key=value`, like below:

Comment (andrehjr) [inline src/doc/build-script.md]: Of course :) Sounds good!

Just updated the commit with your suggestion.

**Comment by alexcrichton** [comment]
@bors: r+ ea0550c1b805379a63cfeeced09bb257e1583cb5

Thanks!

**Pull request merged**